config/config-builder/src/validator_config.rs

```diff
@@ -140,24 +140,19 @@ impl ValidatorConfig {
 
     pub fn build(&self) -> Result<NodeConfig> {
         let mut configs = self.build_set()?;
-        let first_peer_id = configs[0]
-            .validator_network
-            .as_ref()
-            .ok_or(Error::MissingValidatorNetwork)?
-            .peer_id;
-        let mut config = configs.swap_remove(self.index);
 
+        // Extract and format first node's advertised address for the cluster
+        // bootstrap.
+        let seed_peers = self.build_seed_peers(&configs[0])?;
+
+        let mut config = configs.swap_remove(self.index);
```

Comment:
I'd suggest adding a comment here, took me a while to figure out that we are removing ourselves